src/plugins/jwt.ts

```diff
@@ -21,19 +23,28 @@ export default fp((fastify, options, done) => {
   })
 
   fastify.decorate('hasRole', function (roles: ROLE | ROLE[]) {
-    const allowedRoles = Array.isArray(roles) ? roles : [roles]
+    const allowedRoles: ROLE[] = Array.isArray(roles) ? roles : [roles]
 
     return async function (req: FastifyRequest) {
       let payload: JWTPayload
-
       try {
         payload = await req.jwtVerify<JWTPayload>()
       } catch {
-        throw new UnauthorizedError('Not authenticated.')
+        throw new UnauthorizedError('Invalid or missing token.')
       }
 
-      if (!allowedRoles.includes(payload.role))
-        throw new ForbiddenError(`Users with role '${payload.role}' cannot perform this operation.`)
+      const [ user ] = await fastify.db
+        .select()
+        .from(userTable)
+        .where(eq(userTable.id, payload.sub))
+
+      if (!user) {
+        throw new UnauthorizedError('User not found.')
+      }
+
+      if (!allowedRoles.includes(user.role as ROLE)) {
+        throw new ForbiddenError(`User with role '${user.role}' cannot perform this operation.`)
```

Comment:
DO NOT give extra information on a user. If this is a hacker trying things out he does not need to know the role of the user he is trying. Make a generic message like "Permission denied" or something like that.